### DISCUSSION TITLE
Getting type error when using string to invalidate queries.

### DISCUSSION THREAD

**Discussion by bombillazo** [TanStack/query#5788] [Q&A]
Hey, 

I am following the invalidation query docs to invalidate all queries with the term `rates` as the first element in the query key array by using
```ts
  queryClient.invalidateQueries('rates');
```
I am using the useQueryClient hook to get the queryClient.

This clears the queries I want to clear, but I am getting a type error when I use the string:
```ts
No overload matches this call.
  Overload 1 of 2, '(filters?: InvalidateQueryFilters<unknown> | undefined, options?: InvalidateOptions | undefined): Promise<void>', gave the following error.
    Type '"rates"' has no properties in common with type 'InvalidateQueryFilters<unknown>'.
  Overload 2 of 2, '(queryKey?: QueryKey | undefined, filters?: InvalidateQueryFilters<unknown> | undefined, options?: InvalidateOptions | undefined): Promise<void>', gave the following error.
    Argument of type 'string' is not assignable to parameter of type 'readonly unknown[]'.ts(2769)
```
if I try putting it in an array:
```ts
  queryClient.invalidateQueries(['rates']);
```
the type error goes away but the query invalidation does not work. How can i fix this? Thanks

**Comment by jiarrdgr8** [comment]
try doing
`queryClient.invalidateQueries({queryKey: ['rates']});`

**Reply by loktar00** [reply]
That works.. but the docs specifically mention being able to do it with a string. Is this just a TS thing, or do the docs need to be updated?

**Reply by TkDodo** [reply]
where do the docs say this please?

**Reply by Zack-Dx** [reply]
worked for me as well, but what it says something else in the docs?

**Reply by canyouseethroughme** [reply]
If you use React Query v3, in docs it shows ```queryClient.invalidateQueries('rates');``` , but from v4 upwards it shows ```  queryClient.invalidateQueries({ queryKey: ['todos'] })```

**Reply by lucasruy** [reply]
The problem starts when you search `invalidateQueries` on Google, click on the first result of your search and then redirected to the v3 document. Maybe this is causing the misunderstanding.

The latest version doc, it's updated:
- https://tanstack.com/query/latest/docs/framework/react/guides/query-invalidation

**Reply by soni-canra-wiguna** [reply]
how to invalidate multiple queries?

**Reply by UjjwalKaul** [reply]
> try doing `queryClient.invalidateQueries({queryKey: ['rates']});`

thanks bro lifesaver

**Reply by abdulbasit-005** [reply]
thanks it workd
